Removing ParsedAst.Declaration.Relation and ParsedAst.Declaration.Lat…

2022H1030042G opened the pull request:
Refer: #5769

magnus-madsen commented:
- Instead of commenting please just remove the code.
- For the broken tests, you will want to remove the `rel` and `lat` declarations in them.

2022H1030042G commented:
> * Instead of commenting please just remove the code.
> * For the broken tests, you will want to remove the `rel` and `lat` declarations in them.

Okay, now I have passed all the tests and commented and removed the required code. I have pushed the new changes. Please check it once.

magnus-madsen commented:
Need to push :)

2022H1030042G commented:
> Need to push :)

I did it already, https://github.com/flix/flix/commit/ed6a720921e316d6aa15d4e5a33e19edab207541

magnus-madsen commented:
You say the code is removed, but it looks commented to me?

magnus-madsen commented:
> > You say the code is removed, but it looks commented to me?
> 
> Yes, some of it is still commented, just wanted to make sure everything works. Should I remove all of it?

Just remove the code. Its in git and the diff anyway.

2022H1030042G commented:
> > > > You say the code is removed, but it looks commented to me?
> > > 
> > > 
> > > Yes, some of it is still commented, just wanted to make sure everything works. Should I remove all of it?
> > 
> > 
> > Just remove the code. Its in git and the diff anyway.
> 
> Sure, okay the [ed6a720](https://github.com/flix/flix/commit/ed6a720921e316d6aa15d4e5a33e19edab207541) commit changes were not done by me , it's because I pulled the changes from the repository and then committed, that's why it's showing as my commits. I have removed all the comments that I did and going to push it.

FYI, attached is the screenshot of the files that I changed:
![files changed, comments pending to be removed](https://user-images.githubusercontent.com/17452290/234383018-5fc7c09a-e9bb-4d23-9c00-964cdbc17997.png)

magnus-madsen commented:
The example should not be removed, but just refactored by removing the `rel` and `lat` parts.

2022H1030042G commented:
> The examples should not be removed, but just refactored by removing the `rel` and `lat` parts.

If I do not remove the examples, then there will be comments all over the code.

magnus-madsen commented on examples/first-class-constraints-and-fixpoints.flix:
```diff
@@ -1,22 +1,22 @@
-    ParentOf("Pompeia", "Pompey").
-    ParentOf("Sextus", "Pompey").
-}
+///def getFacts(): #{ ParentOf, AncestorOf } = #{
```
For example here, you should change ParentOf to `ParentOf(String, String)` because you have deleted the ParentOf alias above.

magnus-madsen commented:
Let me try to explain what we want to do. If we have:

```
rel ParentOf(x: String, y: String)
rel AncestorOf(x: String, y: String)

/// Returns a collection of facts.
def getFacts(): #{ ParentOf, AncestorOf } = #{
```

then we should think of `rel ParentOf(x: String, y: String)` as introducing an alias that is used in the type of `getFacts`.

Thus the program should be changed to:

```
/// Returns a collection of facts.
def getFacts(): #{ ParentOf(String, String), AncestorOf(String, String) } = #{
```

where you can see that I have inlined the definitions of `ParentOf` and `AncestorOf` and I have removed the `rel`s.

2022H1030042G commented:
How to fix this one for file polymorphic-first-class-constraints.flix?: 

![error](https://user-images.githubusercontent.com/17452290/234575343-7281bb7e-292e-4776-b9e9-f24dceac4125.png)

magnus-madsen commented:
It should be `LabelEdge(Int32)` I think (not square brackets)

magnus-madsen commented:
You can always push to the PR-- no harm in that

2022H1030042G commented on examples/first-class-constraints-and-fixpoints.flix:
```diff
@@ -1,17 +1,14 @@
 
 /// Returns a collection of rules to compute ancestors.
-def getRules(): #{ ParentOf, AncestorOf } = #{
+def getRules(): #{ ParentOf( String, String), AncestorOf( String, String) } = #{
```
What should I replace this with? : Undefined name 'LabelEdge'.

28 | def getRules(): #{ LabelEdge[l], LabelPath[l] } with Order[l] = #{
                        ^^^^^^^^^
                        name not found

2022H1030042G replied:
Please check the recent changes that are pushed (some errors are there in polymorphic file)

2022H1030042G replied:
Can anyone please review?
